File: GameClient/Assets/Scripts/GameLogic/State/GameHallState.cs
using System.Collections;
using System.Collections.Generic;
using UnityEngine;

public class GameHallState : StateBase
{
    public GameHallState()
    {
        currentState = StateType.GameHall;
    }

    public override void Enter()
    {
        Debug.Log("enter GameHall");
        GameHallController.Instance.ShowView();
        GameHallController.Instance.RegisterProto();
    }

    public override void Excute()
    {
        throw new System.NotImplementedException();
    }

    public override void Exit()
    {
        Debug.Log("exit GameHall");
        GameHallController.Instance.CloseView();
        GameHallController.Instance.UnregisterProto();
    }
}

File: GameClient/Assets/HotFix/Lua/FSM/GameHallState.lua
local gameHallState = class(require "FSM/BaseState")

function gameHallState:enter( )
	global.gameHallController:showView()
	global.gameHallController:registerProto()
end

function gameHallState:exit(  )
	global.gameHallController:closeView()
	global.gameHallController:unregisterProto()
end

return gameHallState